### PR TITLE
🐛 📝 re-update all connectors for new CDK version and more stringent javac checks

### PR DESCRIPTION
**Pull request by stephane-airbyte**
This is a no-op in term of features or bugs. This is part 2 of a series of PRs. This builds upon #32118 

I'm only trying to use as many compiler warnings as possible. They're cheap and can help with a lot of bugs.
Unfortunately, until very recently, our CDK was packaged in a way that causes warnings in any package that used it. I fixed that in 0.4.0 but all projects that use local CDK will have a warning, which means we can't turn those into errors.

Since we released a new CDK (0.4.0) that exports everything properly, we now need to use it and fix the connectors that are certified

### CODE REVIEW

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java [hunk 1]
```diff
@@ -148,7 +146,7 @@ public static XminStatus getXminStatus(final JdbcDatabase database) throws SQLEx
    */
   public static Map<AirbyteStreamNameNamespacePair, CursorBasedStatus> getCursorBasedSyncStatusForStreams(final JdbcDatabase database,
                                                                                                           final List<ConfiguredAirbyteStream> streams,
-                                                                                                          final StateManager<AirbyteStateMessage, AirbyteStreamState> stateManager,
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java:151]: StateManager was modified in CDK 0.4.0 to not have any generics. Fixing that here.

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java [hunk 2]
```diff
@@ -39,12 +39,13 @@
  * from table where pk > pk_max_4 order by pk limit 1,800,000. Final query, since there are zero
  * records processed here.
  */
+@SuppressWarnings("try")
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java:42]: close() throws Exception, which triggers a java warning.

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java [hunk 3]
```diff
@@ -39,12 +39,13 @@
...
 public class MySqlInitialLoadRecordIterator extends AbstractIterator<JsonNode>
     implements AutoCloseableIterator<JsonNode> {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MySqlInitialLoadRecordIterator.class);
 
-  private final JdbcCompatibleSourceOperations sourceOperations;
+  private final JdbcCompatibleSourceOperations<MysqlType> sourceOperations;
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java:48]: this was a raw type before

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadHandler.java [hunk 4]
```diff
@@ -49,7 +48,7 @@ public class MySqlInitialLoadHandler {
   private static final long RECORD_LOGGING_SAMPLE_RATE = 1_000_000;
   private final JsonNode config;
   private final JdbcDatabase database;
-  private final JdbcCompatibleSourceOperations sourceOperations;
+  private final MySqlSourceOperations sourceOperations;
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadHandler.java:51]: MySqlSourceOperation is used everywhere in this class, and is not parameterized

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlQueryUtils.java [hunk 5]
```diff
@@ -84,7 +83,7 @@ public static Set<String> getStorageEngines(final JdbcDatabase database, final S
       return storageEngines;
     } catch (final Exception e) {
       LOGGER.info("Storage engines could not be determined");
-      return Collections.EMPTY_SET;
+      return Collections.emptySet();
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlQueryUtils.java:86]: collections.EMPTY_SET is a SET, while Collections.emptySet() returns a Set<>

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlQueryUtils.java [hunk 6]
```diff
@@ -36,16 +36,15 @@ public class MySqlQueryUtils {
...
-       """;
+  public static final String TABLE_ESTIMATE_QUERY = """
+                                                     SELECT
+                                                       (data_length + index_length) as %s,
+                                                       AVG_ROW_LENGTH as %s
+                                                    FROM
+                                                       information_schema.tables
+                                                    WHERE
+                                                       table_schema = '%s' AND table_name = '%s';
+                                                    """;
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlQueryUtils.java:47]: fixing whatespaces that would have been removed by the compiler (and that triggered a warning)

File: airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoUtil.java [hunk 7]
```diff
@@ -254,6 +255,7 @@ private static Set<Field> getFieldsInCollection(final MongoCollection<Document>
 
     try (final MongoCursor<Document> cursor = output.allowDiskUse(true).cursor()) {
       while (cursor.hasNext()) {
+        @SuppressWarnings("unchecked")
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoUtil.java:258]: the get() returns an Object, so there's an unchecked cast here

File: airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoUtil.java [hunk 8]
```diff
@@ -163,6 +163,7 @@ public static Optional<CollectionStatistics> getCollectionStatistics(final Mongo
       try (final MongoCursor<Document> cursor = output.allowDiskUse(true).cursor()) {
         if (cursor.hasNext()) {
           final Document stats = cursor.next();
+          @SuppressWarnings("unchecked")
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoUtil.java:166]: get() returns an Object, so there's an unchecked cast here

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeV1V2Migrator.java [hunk 9]
```diff
@@ -87,9 +87,11 @@ protected Optional<SnowflakeTableDefinition> getTableIfExists(final String names
   @Override
   protected NamespacedTableName convertToV1RawName(final StreamConfig streamConfig) {
     // The implicit upper-casing happens for this in the SqlGenerator
+    @SuppressWarnings("deprecation")
+    String tableName = this.namingConventionTransformer.getRawTableName(streamConfig.id().originalName());
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeV1V2Migrator.java:91]: getRawTableName is deprecated. I had to move the call to its own line for the SuppressWarnings to be accepted

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.java [hunk 10]
```diff
@@ -145,6 +145,7 @@ public boolean existingSchemaMatchesStreamConfig(final StreamConfig stream, fina
             (map, column) -> map.put(column.getKey(), column.getValue().type()),
             LinkedHashMap::putAll);
     // soft-resetting https://github.com/airbytehq/airbyte/pull/31082
+    @SuppressWarnings("deprecation")
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.java:148]: isNullable is deprecated

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeColumnDefinition.java [hunk 11]
```diff
@@ -9,4 +9,11 @@
  * destination-snowflake, where we created PK columns as NOT NULL. This caused a lot of problems
  * because many sources emit null PKs. We may want to remove this field eventually.
  */
-public record SnowflakeColumnDefinition(String type, @Deprecated boolean isNullable) {}
+public record SnowflakeColumnDefinition(String type, boolean isNullable) {
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeColumnDefinition.java:12]: Deprecated on a constuctor was triggering a warning, while the call to isNullable was also triggering a warning on the calling site for deprecation. We're keeping the deprecation, but removing the warning in this class

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/uploader/BigQueryUploaderFactory.java [hunk 12]
```diff
@@ -44,7 +44,7 @@ public class BigQueryUploaderFactory {
                                                    "Destination Default" option.
 
                                                  More details:
-                                                   """;
+                                                 """;
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/uploader/BigQueryUploaderFactory.java:47]: fixing whitespace that would be removed by the compiler but triggered a warning

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordStandardConsumer.java [hunk 13]
```diff
@@ -20,6 +20,7 @@
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
+@SuppressWarnings("try")
 public class BigQueryRecordStandardConsumer extends AsyncStreamConsumer {
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordStandardConsumer.java:24]: close throws Exception, hence the suppresswarnings

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordConsumer.java [hunk 14]
```diff
@@ -30,7 +30,8 @@
 /**
  * Record Consumer used for STANDARD INSERTS
  */
-public class BigQueryRecordConsumer extends FailureTrackingAirbyteMessageConsumer implements AirbyteMessageConsumer {
+@SuppressWarnings("try")
+class BigQueryRecordConsumer extends FailureTrackingAirbyteMessageConsumer implements AirbyteMessageConsumer {
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordConsumer.java:34]: close() throws Exception, hence the SuppressWarnings

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryGcsOperations.java [hunk 15]
```diff
@@ -160,6 +160,7 @@ public void copyIntoTableFromStage(final String datasetId,
   }
 
   @Override
+  @Deprecated
   public void cleanUpStage(final String datasetId, final String stream, final List<String> stagedFiles) {
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryGcsOperations.java:164]: the overloaded cleanUpStage is deprecated,so this one needs to be annotated too

**Review by rodireich — APPROVED** [review]
0 warnings!
Looks good to me. 

Maybe just consult with @airbytehq/destinations since a bunch of their connectors are upgraded as well.

File: docs/integrations/destinations/snowflake.md [hunk 16]
```diff
@@ -277,6 +277,7 @@ Otherwise, make sure to grant the role the required permissions in the desired n
 
 | Version         | Date       | Pull Request                                               | Subject                                                                                                                                                         |
 |:----------------|:-----------|:-----------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 3.4.6           | 2023-11-03 | [32125](https://github.com/airbytehq/airbyte/pull/32125)   | Fix compilation warnings 7                                                                                                                                 |
```
Comment (postamar) [inline docs/integrations/destinations/snowflake.md:280]: typo

File: docs/integrations/destinations/bigquery.md [hunk 17]
```diff
@@ -140,6 +140,7 @@ Now that you have set up the BigQuery destination connector, check out the follo
 
 | Version | Date       | Pull Request                                               | Subject                                                                                                                                                         |
 |:--------|:-----------|:-----------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 2.3.6   | 2023-11-03 | [32125](https://github.com/airbytehq/airbyte/pull/32125)   | fix compilation warnings                                                                                                                                         |
```
Comment (postamar) [inline docs/integrations/destinations/bigquery.md:143]: nit: here and in the other changelogs, I think the significant change is rather that we're adopting version 0.4.0 of the CDK.

I'm assuming that these changelogs are read when the connector's misbehaving after an upgrade; knowing about the CDK change is probably going to be more relevant in that use case.

**Review by postamar — CHANGES_REQUESTED** [review]
Can you wait until https://github.com/airbytehq/airbyte/pull/32050 merges please? I expect it to pass CI in an hour or so, since it touches quite a few of the same connectors. It bumps them to 0.4.0 as well.

Blocking the merge is the only reason why I'm "requesting changes"

edit: incidentally, my PR fixes the destination-s3 build

**Review by postamar — APPROVED** [review]
I've removed myself as a blocker spinning off the connector changes in https://github.com/airbytehq/airbyte/pull/32050 into separate PRs:
https://github.com/airbytehq/airbyte/pull/32173
https://github.com/airbytehq/airbyte/pull/32174
https://github.com/airbytehq/airbyte/pull/32175
https://github.com/airbytehq/airbyte/pull/32176
https://github.com/airbytehq/airbyte/pull/32177
https://github.com/airbytehq/airbyte/pull/32178
https://github.com/airbytehq/airbyte/pull/32179
https://github.com/airbytehq/airbyte/pull/32180

Not all have release notes, because the CDK bump is often a functional no-op.

Please feel free to do whatever suits you best with these. I set them to auto-merge before signing off.

**Review by postamar — APPROVED** [review]

**Pull request merged**